src/scripts/DataLayer.js

```diff
@@ -313,21 +313,41 @@ DataLayer.Manager.prototype._isMatching = function(listener, item) {
 
   if (item.type === DataLayer.constants.itemType.DATA) {
     if (listenerConfig.on === DataLayer.constants.event.CHANGE) {
-      isMatching = true;
+      isMatching = this._isSelectorMatching(listenerConfig, itemConfig);
     }
   } else if (item.type === DataLayer.constants.itemType.EVENT) {
     if (listenerConfig.on === DataLayer.constants.event.EVENT ||
       listenerConfig.on === itemConfig.event) {
-      isMatching = true;
+      isMatching = this._isSelectorMatching(listenerConfig, itemConfig);
     }
     if (itemConfig.data &&
       listenerConfig.on === DataLayer.constants.event.CHANGE) {
-      isMatching = true;
+      isMatching = this._isSelectorMatching(listenerConfig, itemConfig);
     }
   }
   return isMatching;
 };
 
+/**
+ * Checks if the given listerConfig has a selector and if so, is it matching with the data payload of the itemConfig.
+ *
+ * @param {DataLayer.Item} listenerConfig Config of the listener to extract the selector from.
+ * @param {DataLayer.Item} itemConfig The item config withe data.
+ * @returns {Boolean} true if no selector is provided or the given selector is matching, false otherwise.
```

Comment:
I'd rephrase it e.g.:
true if a selector is not provided or if the given selector is matching, false otherwise.